Fix local_cache in multi_returner

a76yyyy opened the pull request:
### What does this PR do?

Fixes incorrect storage of `_return`-type job results in local_cache when using `multi_returner` configuration. Now strictly follows the logic in `salt.utils.job.store_job` by only executing multi-returner storage when `load['cmd']` is not equal to `'_return'`.

### What issues does this PR fix or reference?

None

### Previous Behavior

With multi-returner configuration:

```yaml
master_job_cache: multi_returner
multi_returner:
  - local_cache
  - elasticsearch # or other
```

All types of return data (including `cmd: _return` job results) were stored in `local_cache`, violating the expected behavior defined in store_job logic:

https://github.com/saltstack/salt/blob/f906ca5052ed90245538617c14d7f2dbc4b72127/salt/utils/job.py#L127-L130

### New Behavior

Now only save_load when `returner_ == "local_cache" and load['cmd'] != '_return'`

### Merge requirements satisfied?

**[NOTICE] Bug fixes or features added to Salt require tests.**

<!-- Please review the test documentation for details on how to implement tests
into Salt's test suite:
https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html -->
- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
No

a76yyyy committed:
Fix local_cache in multi_returner

twangboy reviewed (changes requested):
Please add a changelog and some tests

a76yyyy committed:
Add changelog

twangboy reviewed (changes requested):
Please write a test for this

dwoz commented:
> Please write a test for this

@a76yyyy

Commit:
Fix get_jids_filter typo and save_reg early-return in multi_returner

The ``get_jid_filter`` (singular) name introduced earlier in this PR
did not match Salt's canonical ``<returner>.get_jids_filter`` (plural)
that the ``salt/runners/jobs.py`` runner looks up. Both the function
name and the ``fstr`` lookup are renamed to ``get_jids_filter`` so the
runner integration works end-to-end.

``save_reg`` was ``return``-ing inside the loop, writing to only the
first configured returner that implemented ``save_reg`` and silently
skipping the rest. The early ``return`` is dropped so every configured
returner is written, matching the write-to-all contract that
``returner()``, ``save_load()``, ``save_minions()``, ``update_endtime()``
and friends already follow.

Regression test added at
``tests/pytests/unit/returners/test_multi_returner.py`` covering:

* ``save_load`` skips ``local_cache`` for ``cmd == '_return'`` and
  dispatches normally otherwise (the original fix in this PR);
* ``save_load`` forwards the ``minions`` argument (previously dropped);
* ``get_jids_filter`` dispatches and merges under the canonical plural
  name, and the misnamed singular attribute does not exist;
* ``save_reg`` writes to every configured returner that implements it.

All seven new tests pass on this branch and fail on ``origin/master``.

Co-authored-by: a76yyyy <q981331502@163.com>

twangboy reviewed (changes requested):
Looks great. Please create against the earliest branch where this bug exists.
